src/pkgcore/ebuild/ebuild_src.py

```diff
@@ -252,6 +252,8 @@ def eapi(self):
             if (mo := _EAPI_str_regex.match(line)) and (eapi_str := mo.group('EAPI')):
                 eapi = _EAPI_regex.match(line).group('EAPI')
             break
+        if hasattr(i, "close"):
```

Comment:
Is there a case that it wouldn't have the `close` method? It feels weird to check for that.

Reply:
It does feel weird. The `i` variable is either

```python
i = fileutils.readlines_utf8(ebuild.path)
```
or
```python
i = (x.strip() for x in ebuild.text_fileobj())
```
(I've only seen the first variant so far).

In the latter case, does `ebuild.text_fileobj()` need to be closed?

Reply:
Probably.

Reply:
OK, reworked to call close() for the fileutls.readlines_utf8(), but I've skipped the other variant, simply because I haven't seen it executed yet :-)